src/Event/EventManager.php

```diff
@@ -442,6 +442,24 @@ public function prioritisedListeners($eventKey)
     }
 
     /**
+     * Returns the listeners matching a specified pattern
+     *
+     * @param string $eventKeyPattern Pattern to match.
+     * @return array
+     */
+    public function matchingListeners($eventKeyPattern)
+    {
+        $matchPattern = '/' . preg_quote($eventKeyPattern, "/") . '/';
```

Comment:
@markstory Well, based on the #4192 request, there was the idea of a robust pattern matching. For example, the following was mentioned in a comment:

    // Check with trailing wildcards
    $events->hasListener('Model.*');

So with my implementation using the ```preg_grep()``` and not providing any anchors I could locate 'test' in both of the EventManagerTest.php test-case scenarios below. It would enable the developer to find both the 'test.event' and 'event.test' listeners. This was before the introduction yesterday of the ```preg_quote()``` function call to escape all regex from the $eventKeyPattern argument.  Before that addition I could pass an achor as part of the pattern and match it accordingly; however -- as you said previously -- we can expand this over time if necessary.  Let me know what you think!

    public function testMatchingListeners()
    {
        $manager = new EventManager();
        $manager->attach('fakeFunction1', 'fake.event');
        $manager->attach('fakeFunction2', 'real.event');
        $manager->attach('fakeFunction3', 'test.event');
        $manager->attach('fakeFunction4', 'event.test');

        $this->assertArrayHasKey('fake.event', $manager->matchingListeners('fake'));
        $this->assertArrayHasKey('real.event', $manager->matchingListeners('real'));
        $this->assertArrayHasKey('test.event', $manager->matchingListeners('test'));
        $this->assertArrayHasKey('event.test', $manager->matchingListeners('test'));